app/models/XPath.scala

```diff
@@ -33,13 +37,41 @@ case class XPath(value: String) {
     "Consignment"
   )
 
+  val preTaskList: JourneyTask = TraderDetails
+
   def isAmendable: Boolean = {
     val regex = "^/CC015C/(.+)$".r
     value match {
       case regex(section) if sections.exists(section.startsWith) => true
       case _                                                     => false
     }
   }
+
+  def sectionError: Option[(String, Status.Value)] =
+    this.task match {
+      case Some(task) => Some((task.taskName, Status(Status.Error.id)))
+      case _          => None
+    }
+
+  def task: Option[JourneyTask] = {
+    val pf: PartialFunction[String, JourneyTask] = _.replace("/CC015C/", "") match {
+      case x if x.matches("^TransitOperation/declarationType$")          => PreTaskList
+      case x if x.matches("^TransitOperation/tirCarnetReference$")       => PreTaskList
+      case x if x.matches("^TransitOperation/securityDetailsType$")      => PreTaskList
```

Comment:
Be careful not to get mixed up between the user answers paths and the xpaths. If the xsds are to be believed these should be `TIRCarnetNumber` and `security` (https://github.com/hmrc/manage-transit-movements-departure-cache/blob/780d954d67a2af39fabbb3676492ed6ad3fd56cf/conf/xsd/ctypes.xsd#L19179 and https://github.com/hmrc/manage-transit-movements-departure-cache/blob/780d954d67a2af39fabbb3676492ed6ad3fd56cf/conf/xsd/ctypes.xsd#L19199) cc @Reece-Carruthers @MatthewAlexanderBeck